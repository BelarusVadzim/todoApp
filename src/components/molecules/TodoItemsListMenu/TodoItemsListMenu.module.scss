@import 'src/variables.scss';

.menuItem {
  display: flex !important;
  flex: 1 1;
  align-items: center;
  justify-content: center;
  font-size: $todoItemsListMenu-font-size !important;
}

.todoItemsListMenu {
  height: $todoItemsListMenu-height;
  vertical-align: middle;
  display: flex;
  justify-content: space-around;
}

.label {
  justify-content: left;
  margin-left: 1.5rem;

  @extend .menuItem;
}

.filterSection {
  min-width: 7.5rem;
  justify-content: space-evenly;

  @extend .menuItem;
}

.buttonArea {
  justify-content: right;
  margin-right: 1.5rem;

  @extend .menuItem;
}

.button {
  font-size: $todoItemsListMenu-font-size !important;
}

