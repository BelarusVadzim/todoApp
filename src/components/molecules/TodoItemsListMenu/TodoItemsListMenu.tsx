import React from 'react';
import { Button } from 'components/atoms';
import style from './TodoItemsListMenu.module.scss';
import { TodoItemsListFilterSection } from '../TodoItemsListFilterSection';
import { useTodoStateService } from 'hooks';

export const TodoItemsListMenu: React.FC = () => {
  const { todos, dispatchCompletedTodosDeleted } = useTodoStateService();
  const buttonClearClick = () => dispatchCompletedTodosDeleted();

  return (
    <div className={style.todoItemsListMenu}>
      <div className={style.itemsLeftLabel}>{todos.length} items left</div>
      <TodoItemsListFilterSection className={style.filterSection} />
      <div className={style.buttonArea}>
        <Button
          value="Clear Completed"
          onClick={buttonClearClick}
          className={style.button}
        />
      </div>
    </div>
  );
};
