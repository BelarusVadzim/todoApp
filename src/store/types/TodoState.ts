import { TodoNote } from 'types';

export type TodoState = {
  todos: TodoNote[];
  filter?: string;
  pending?: boolean;
};
