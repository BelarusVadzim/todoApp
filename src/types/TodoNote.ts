export type TodoNote = {
  id?: number;
  done?: boolean;
  text: string;
};
