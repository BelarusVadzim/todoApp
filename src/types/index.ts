export * from './PropsWithClassName';
export * from './TodoNote';
