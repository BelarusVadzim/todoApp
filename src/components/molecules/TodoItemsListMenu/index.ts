import TodoItemsListMenu from './TodoItemsListMenu';

export { TodoItemsListMenu };