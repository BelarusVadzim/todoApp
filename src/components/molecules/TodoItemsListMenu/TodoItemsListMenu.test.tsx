import { render, screen } from '@testing-library/react';
import TodoItemsListMenu from './TodoItemsListMenu';
import { TodoNote } from 'types';
import userEvent from '@testing-library/user-event';

const dispatchCompletedTodosDeleted = jest.fn();
const clearCompletedButtonTestId = 'clear-completed-button-test-id';
const todo1 = { text: 'todo1' };
const todo2 = { text: 'todo2' };
const todos: TodoNote[] = [todo1, todo2];

jest.mock('hooks', () => ({
  useTodoStateService: () => ({ todos, dispatchCompletedTodosDeleted }),
}));

jest.mock('components/atoms', () => ({
  TextPlace: jest.fn(({ className, children }) => {
    return <div className={className}>{children}</div>;
  }),
  Button: jest.fn(({ value, className, onClick }) => {
    return (
      <div
        data-testid={clearCompletedButtonTestId}
        className={className}
        onClick={onClick}
      >
        {value}
      </div>
    );
  }),
}));

jest.mock('../TodoItemsListFilterSection', () => ({
  TodoItemsListFilterSection: jest.fn(({ className }) => {
    return <div className={className} />;
  }),
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('<TodoItemsListMenu />', () => {
  it('should render properly', () => {
    const component = render(<TodoItemsListMenu />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when Clear Completed button click should call expected functions with expected values', () => {
    render(<TodoItemsListMenu />);

    const inputElement = screen.getByTestId(clearCompletedButtonTestId);

    userEvent.click(inputElement);

    expect(dispatchCompletedTodosDeleted).toBeCalledTimes(1);
  });
});
