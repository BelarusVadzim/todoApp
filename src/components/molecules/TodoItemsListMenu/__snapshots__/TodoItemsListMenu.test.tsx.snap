// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItemsListMenu /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="todoItemsListMenu"
  >
    <div
      class="label"
    >
      2 items left
    </div>
    <div
      class="filterSection"
    />
    <div
      class="button"
      data-testid="clear-completed-button-test-id"
    >
      Clear Completed
    </div>
  </div>
</DocumentFragment>
`;
